plugins/manipulator.cpp

```diff
@@ -542,94 +782,782 @@ class StringFormatter {
     T_optlist opt_list;
 };
 
-namespace unit_ops {
-    string get_real_name(UnitInfo *u)
-        { return Translation::TranslateName(&u->unit->name, false); }
-    string get_nickname(UnitInfo *u)
-        { return Translation::TranslateName(Units::getVisibleName(u->unit), false); }
-    string get_real_name_eng(UnitInfo *u)
-        { return Translation::TranslateName(&u->unit->name, true); }
-    string get_nickname_eng(UnitInfo *u)
-        { return Translation::TranslateName(Units::getVisibleName(u->unit), true); }
-    string get_first_nickname(UnitInfo *u)
-    {
-        return Translation::capitalize(u->unit->name.nickname.size() ?
-            u->unit->name.nickname : u->unit->name.first_name);
-    }
-    string get_first_name(UnitInfo *u)
-        { return Translation::capitalize(u->unit->name.first_name); }
-    string get_last_name(UnitInfo *u)
-    {
-        df::language_name name = u->unit->name;
-        string ret = "";
-        for (int i = 0; i < 2; i++)
+static bool labor_skill_map_made = false;
+
+namespace unit_info_ops{
+
+    //these are copied from managelabor.cpp ( skill_attr_weight )
```

Comment:
labormanager? Might be better to move to core if multiple things need it, but not too necessary.